### PR TITLE
zjson: handle aliases on record types

### LINKED ISSUE #758: Alias handling bug in zjson writer
The zjson writer handles aliases to primitive types ok, but not aliases to container types. (The zjson reader likely has a similar issue).

```sh
$ cat alias.tzng 
#0:record[host:ip]
0:[127.0.0.2;]
#myrec=record[host:ip]
#1:record[foo:myrec]
1:[[127.0.0.2;]]
19:21 ~/work/brim/zq(zeekstrings-to-zeekio)
$ zq -f zjson alias.tzng 
{"id":23,"type":[{"name":"host","type":"ip"}],"values":["127.0.0.2"]}
malformed zng value
```

### PR DESCRIPTION
**Pull request by henridf**
closes #758

### CODE REVIEW

File: zio/zjsonio/reader.go [hunk 1]
```diff
@@ -239,7 +239,7 @@ func decodeContainer(builder *zcode.Builder, typ zng.Type, body []interface{}) e
 		// each column either a string value or an array of string values
 		if column == nil {
 			// this is an unset column
-			if zng.IsContainerType(childType) || zng.IsContainerType(columns[k].Type) {
+			if zng.IsContainerType(zng.AliasedType(childType)) || zng.IsContainerType(zng.AliasedType(columns[k].Type)) {
```
Comment (mattnibs) [inline zio/zjsonio/reader.go:242]: being that we don't seem to care about the alias types here I would do this:
- line 273: `childType = zng.AliasedType(childType)`
- line 253: `childType = zng.AliasedType(childType)`
- drop changes on line 242 and 255 and all references to `realType`.
The variable name `realType` had me confused with whether it is referring to the container Type or the child Type.

File: zio/zjsonio/stream.go [hunk 2]
```diff
@@ -108,7 +108,8 @@ func encodeContainer(typ zng.Type, val []byte) (interface{}, error) {
 				childType = columns[k].Type
```
Comment (mattnibs) [inline zio/zjsonio/stream.go:108]: Being that we don't care about aliases at this point I would just on line 94 do `childType = zng.AliasedType(childType) and here: 
```suggestion
				childType = zng.AliasedType(columns[k].Type)
```

File: zio/zjsonio/reader.go (hunk 1 above)
Comment (henridf) [reply zio/zjsonio/reader.go:242]: > being that we don't seem to care about the alias types here I would do this:

Why don't we need it here? I thought we needed to undo aliases wherever we handled types (as we do elsewhere, for example in tzngio). The whole alias scheme remains a bit fuzzy to me though, so I'd be happy to learn here.

> The variable name realType

Ok, I borrowed this name from tzngio, but agree it's not highly descriptive. I'll change the name before merging.

File: zio/zjsonio/stream.go (hunk 2 above)
Comment (henridf) [reply zio/zjsonio/stream.go:108]: I don't think I understand this comment as calling AliasedType in the two places you suggest is functionally identical to calling it in the single place that it is currently called?

File: zio/zjsonio/reader.go (hunk 1 above)
Comment (henridf) [reply zio/zjsonio/reader.go:242]: > > being that we don't seem to care about the alias types here I would do this:

> Why don't we need it here? I thought we needed to undo aliases wherever we handled types (as we do elsewhere, for example in tzngio). The whole alias scheme remains a bit fuzzy to me though, so I'd be happy to learn here.


@mattnibs I added a test in 0f41a8b2ce22c82c81fed686327dd36464c243c5 that answers this question - without the changes suggested in https://github.com/brimsec/zq/pull/793#discussion_r426793717, this test fails.

File: zio/zjsonio/stream.go (hunk 2 above)
Comment (mattnibs) [reply zio/zjsonio/stream.go:108]: I didn't like the name realType- with your change below this isn't needed.

**Pull request merged**